dash/include/dash/HaloMatrix.h

```diff
@@ -0,0 +1,259 @@
+#ifndef DASH__HALOMATRIX_H_INCLUDED
+#define DASH__HALOMATRIX_H_INCLUDED
+
+#include <dash/dart/if/dart.h>
+
+//#include <dash/Team.h>
+#include <dash/Pattern.h>
+//#include <dash/GlobRef.h>
+#include <dash/GlobMem.h>
+//#include <dash/Allocator.h>
+#include <dash/iterator/HaloMatrixIterator.h>
+#include <dash/Halo.h>
+#include <dash/Matrix.h>
+#include <type_traits>
+
+namespace dash {
+
+template<typename MatrixT, typename HaloSpecT>
+class HaloMatrix
```

Comment:
I prefer moving it to `dash/experimental`, as already two examples are using the halo stuff. `dash/experimental` won't be included in `libdash.h` so whenever a user dares to use it, he has to include ` dash/experimental/...` explicitly.
Furthermore moving it to examples will give name clashes in the future.